### PR TITLE
`azurerm_spring_cloud_gateway_route_config` - support for the `protocol` property

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_gateway_route_config_resource.go [hunk 1]
```diff
@@ -64,6 +64,16 @@ func resourceSpringCloudGatewayRouteConfig() *pluginsdk.Resource {
 				},
 			},
 
+			"protocol": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(appplatform.GatewayRouteConfigProtocolHTTP),
```
Comment (tombuildsstuff) [inline internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:70]: this is an insecure default, what's the reasoning for making this `HTTP` rather than `HTTPS` (or Required?)

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @ms-henglu 

Thanks for this PR - I've taken a look through and whilst this PR looks good on the whole I think we need to think about the default value here, using `HTTP` as a default rather than `HTTPS` (or making this field Required, or not having a default value) is an insecure default -as such we probably need to change this default value.

Thanks!

File: internal/services/springcloud/spring_cloud_gateway_route_config_resource.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:70]: The server-side's default value is `HTTP`, if we make it defaults to `HTTPS` or required field. It will introduce breaking change.

Comment (stephybun) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go:70]: Can we make this field Required for 4.0 then and also add a note in the docs warning users that this is insecure by default.

Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_gateway_route_config_resource.go]: Thanks @stephybun , I've updated the PR, please take another look.

File: website/docs/r/spring_cloud_gateway_route_config.html.markdown [hunk 2]
```diff
@@ -74,6 +75,10 @@ The following arguments are supported:
 
 * `open_api` - (Optional) One or more `open_api` blocks as defined below.
 
+* `protocol` - (Optional) Specifies the protocol of routed Spring Cloud App. Allowed values are `HTTP` and `HTTPS`. Defaults to `HTTP`. 
+
+~> **Note:** `protocol`'s default value `HTTP` is insecure.
```
Comment (stephybun) [inline website/docs/r/spring_cloud_gateway_route_config.html.markdown:80]: Can we rephrase this to 
```suggestion
~> **Note:** You likely want to use `HTTPS` in a production environment, since `HTTP` offers no encryption.
```

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 🎺
